### ISSUE TITLE
direct "remove" and "mark as unread" buttons

### ISSUE THREAD

**Issue by zordsdavini** [nextcloud/mail#5775]
It would be grate to have configurable from settings option to show direct buttons "remove" and "mark as unread" in message list and message preview. Now clicking drop-down to select action triples your time and you lose concentration.

**Comment by GretaD** [comment]
Hello @zordsdavini, thank you for your suggestion.

Unfortunately our UI doesnt allow us to add more actions in a message as it would be overcrowded. There have been couple of discussion on how we can improve that and maybe in the future we can see how to simplify it. 

For now i would suggest you to use shortcut, 'u' for unread and 'delete' button for delete.

**Comment by zordsdavini** [comment]
@GretaD  As I said it could be under settings. Because in my display I have plenty of space. If it is complicated on mobiles - just hide tham but that could go under settings, too. Let the people decide how they want to see it.
Thank you for shortcuts - I use them but sometimes we are to lazy and holding only mouse or using touchscreens

**Comment by ChristophWurst** [comment]
@nimishavijay @jancborchardt your call :)

**Comment by jancborchardt** [comment]
We could indeed think about expanding certain actions from the menu if the space/width allows it, just like we do in the mockups for the multiselect improvements: https://github.com/nextcloud/mail/issues/6070

And as @zordsdavini says, "Delete" and "Mark unread" are good candidates for that, since next to "Reply" they cover the [4 Ds of time management](https://www.productplan.com/glossary/4-ds-of-time-management/):
- Do: Reply
- Defer: Mark unread
- Delegate: (Sort of reply as well, minorly "Forward", but doesn’t need to show as it’s not as common)
- Delete: Delete

So the order of actions could be
- "Reply" as primary action with text
- "Mark unread" as secondary
- "Delete" as secondary or tertiary
- More actions menu as icon

@nimishavijay what do you think? Depending on the width we could show "Mark unread" and "Delete" as icons only as well.

**Comment by nimishavijay** [comment]
Good idea! There are quick actions in almost every email client!

I have a couple of questions:
- For the actions inside a message, are there any distinctions between thread-level and message-level actions? For eg. are you more likely to mark the whole thread as unread or just one message? Depending on that we could either have these actions in the header (next to the subject of the thread) or in each message next to the "Reply" button
- Do we need something similar in the message list as well? We could show "Mark unread" and maybe "Delete" on the left of the 3-dot menu for each message on hover. This might change the items in the action menu, especially the 3 "most common" actions in the top in https://github.com/nextcloud/mail/issues/5956

**Comment by handkerchief333** [comment]
My two cents. The most frequently used actions in my emails are:
- Delete
- Reply
- Move

So it would be great if “Move” was also available.

One possibility would be to make it similar to Gmail, when you hover over a preview message, the quick action buttons appear. So this method should guarantee enough space. Of course it would be great if you could define yourself which ones should appear.

![Image](https://github.com/user-attachments/assets/726d4fdf-02f4-48f8-821f-be6d5b05b849)

**Comment by GretaD** [comment]
btw @zordsdavini does our new design for selectin an evelope fixes this issue. If you click the avatar of any envelope, a shortcut menu will be shown:

![Image](https://github.com/user-attachments/assets/c3312841-0e52-4611-b5ae-e9d2841fb3db)

**Comment by handkerchief333** [comment]
> btw [@zordsdavini](https://github.com/zordsdavini) does our new design for selectin an evelope fixes this issue. If you click the avatar of any envelope, a shortcut menu will be shown:

@GretaD This feature is great and I'm using this with the "control" and "click" combo, or "shift". BUT you have to do at least 2 clicks. And the reason for this issue from my point of view is that you can delete a message directly with 1 click. This saves many, many clicks over the course of the day and weeks. It may not sound like it, but in practice there is a big difference in productivity.

**Comment by zordsdavini** [comment]
> btw [@zordsdavini](https://github.com/zordsdavini) does our new design for selectin an evelope fixes this issue. 

This is a little bit different way from first idea but after action icons has been put in message content side and you can do action with one click - this bulk action is very good addition. I like it. I think issue can be closed

**Comment by handkerchief333** [comment]
@zordsdavini @GretaD Maybe I don't understand you correctly, but it's not possible right now with just one click. You have to select the message first, then you can go with the shortcut menu OR in the content side of the message with the action icons (only if you have enough space, wide monitor). Anyway, it's two clicks. You cannot one-click-delete a message in the message (preview) list, which would be a time saver.

**Comment by GretaD** [comment]
> [@zordsdavini](https://github.com/zordsdavini) [@GretaD](https://github.com/GretaD) Maybe I don't understand you correctly, but it's not possible right now with just one click. You have to select the message first, then you can go with the shortcut menu OR in the content side of the message with the action icons (only if you have enough space, wide monitor). Anyway, it's two clicks. You cannot one-click-delete a message in the message (preview) list, which would be a time saver.

yes, thats true, its not one click. But because we already added the shortcut after selecting the message and because of the space issue, there is no way we are going to have another "permanent" shortcut without redesigning the whole envelope list.

I can keep the ticket open, or you can open a new one to improve the shortcut experience, since this ticket is a bit outdated.

**Comment by handkerchief333** [comment]
@GretaD Thank you for clarification. I don't think a whole redesign is needed, "just" these two steps:
1. Output the content action icons of a mail also in the message preview list. But hide them with CSS (and maybe pos. absolute).
2. On mouse hover over the preview message/list, show the action icons, like described in https://github.com/nextcloud/mail/issues/5775#issuecomment-2847077317

I can create another ticket for this feature request if you suggest. Thanks anyway for your contribution and hard work.

**Comment by zordsdavini** [comment]
> [@zordsdavini](https://github.com/zordsdavini) [@GretaD](https://github.com/GretaD) Maybe I don't understand you correctly, but it's not possible right now with just one click. You have to select the message first, then you can go with the shortcut menu OR in the content side of the message with the action icons (only if you have enough space, wide monitor). Anyway, it's two clicks. You cannot one-click-delete a message in the message (preview) list, which would be a time saver.

@handkerchief333 it's depends on how you work. Yes, I have wider screen and I even didn't notice what it hides on smaller ones. But mostly I have opened split view with content on the right and it shows trash icon. I click once on it and next message renders :) so, in this case it is "one" click. At first it was idea to copy gmail functionality
